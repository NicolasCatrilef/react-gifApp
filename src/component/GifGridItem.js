import React from 'react'

export const GifGridItem = ( { id, title, url } ) => {
    
    // console.log( id, title, url );
    
    return (
        <div className = "card animate__animated animate__bounce">
            {/* { img.titule } */}
            <img src = { url } alt = { title } />
           <p> { title } </p> 
        </div>
    )
}
