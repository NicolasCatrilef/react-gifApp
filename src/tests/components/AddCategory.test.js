import { shallow } from "enzyme"
import { AddCategory } from "../../component/AddCategory"
import '@testing-library/jest-dom'

describe('Pruebas AddCategory', () => {
    
    const setCategory = jest.fn();
    let wrapper = shallow( <AddCategory setCategories = { setCategory } /> );

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories = { setCategory } /> );
    } );

    test('Mostrar correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
    });

    test('Prueba sobre Input', () => {
        
        const input = wrapper.find('input');
        const value = 'Hola Mundo';
        
        input.simulate( 'change', { 
            target:  {
                value: value
            }
        } );

        expect( wrapper.find('h2').text().trim() ).toBe( value );
    });
    
    test('Prueba Submit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategory ).not.toHaveBeenCalled();
    });

    test('Llamar setCateogries', () => {
        const value = 'Hola Mundo';
        const input = wrapper.find('input');
        
        input.simulate( 'change', { 
            target:  {
                value: value
            }
        } );

        wrapper.find('form').simulate('submit', { preventDefault(){} });
        expect( setCategory ).toHaveBeenCalled();
        expect( setCategory ).toHaveBeenCalledTimes(1);
        expect( setCategory ).toHaveBeenCalledWith( expect.any( Function ) );
        expect( wrapper.find('input').prop('value') ).toBe('');
    })
    
    
})
