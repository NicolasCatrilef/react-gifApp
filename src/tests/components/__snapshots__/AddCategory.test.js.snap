// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas AddCategory Mostrar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
     
     
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
